src/sonic-yang-models/yang-models/sonic-vlan.yang

```diff
@@ -2,242 +2,255 @@ module sonic-vlan {
-					}
-				}
-
-				leaf ip-prefix {
-					type union {
-						type stypes:sonic-ip4-prefix;
-						type stypes:sonic-ip6-prefix;
-					}
-				}
-
-				leaf scope {
-					type enumeration {
-						enum global;
-						enum local;
-					}
-				}
-
-				leaf family {
-
-					/* family leaf needed for backward compatibility
-					   Both ip4 and ip6 address are string in IETF RFC 6021,
-					   so must statement can check based on : or ., family
-					   should be IPv4 or IPv6 according.
-					 */
-
-					must "(contains(../ip-prefix, ':') and current()='IPv6') or
-						(contains(../ip-prefix, '.') and current()='IPv4')";
-					type stypes:ip-family;
-				}
-			}
-			/* end of VLAN_INTERFACE_LIST */
-		}
-		/* end of VLAN_INTERFACE container */
-
-		container VLAN {
-
-			description "VLAN part of config_db.json";
-
-			list VLAN_LIST {
-
-				key "name";
-
-				leaf name {
-					type string {
-						pattern 'Vlan([0-9]{1,3}|[1-3][0-9]{3}|[4][0][0-8][0-9]|[4][0][9][0-4])';
-					}
-				}
-
-				leaf vlanid {
-					type uint16 {
-						range 1..4094;
-					}
-				}
-
-				leaf alias {
-					type string;
-				}
-
-				leaf description {
-					type string {
-						length 1..255;
-					}
-				}
-
-				leaf-list dhcp_servers {
-					description "Configure the dhcp v4 servers";
-					type inet:ip-address;
-				}
-
-				leaf-list dhcpv6_servers {
-					description "Configure the dhcp v6 servers";
-					type inet:ipv6-address;
-				}
-
-				leaf mtu {
-					type uint16 {
-						range 1..9216;
-					}
-				}
-
-				leaf admin_status {
-					type stypes:admin_status;
-				}
-
-				leaf mac {
-					type yang:mac-address;
-				}
-			}
-			/* end of VLAN_LIST */
-		}
-		/* end of container VLAN */
-
-		container VLAN_MEMBER {
-
-			description "VLAN_MEMBER part of config_db.json";
-
-			list VLAN_MEMBER_LIST {
-
-				key "name port";
-
-				leaf name {
-					type leafref {
-						path "/vlan:sonic-vlan/vlan:VLAN/vlan:VLAN_LIST/vlan:name";
-					}
-				}
-
-				leaf port {
-					/* key elements are mandatory by default */
-					type union {
-						type leafref {
-							path "/port:sonic-port/port:PORT/port:PORT_LIST/port:name";
-						}
-						type leafref {
-							path "/lag:sonic-portchannel/lag:PORTCHANNEL/lag:PORTCHANNEL_LIST/lag:name";
-						}
-					}
-				}
-
-				leaf tagging_mode {
-					mandatory true;
-					type stypes:vlan_tagging_mode;
-				}
-			}
+    namespace "http://github.com/Azure/sonic-vlan";
+    prefix vlan;
+
+    import ietf-inet-types {
+        prefix inet;
+    }
+
+    import ietf-yang-types {
+        prefix yang;
+    }
+
+    import sonic-types {
+        prefix stypes;
+    }
+
+    import sonic-extension {
+        prefix ext;
+    }
+
+    import sonic-port {
+        prefix port;
+        revision-date 2019-07-01;
+    }
+
+    import sonic-portchannel {
+        prefix lag;
+    }
+
+    import sonic-vrf {
+        prefix vrf;
+    }
+
+    description "VLAN yang Module for SONiC OS";
+
```

Comment:
We need the new revision for the SAG support.

Reply:
Add revision in updated commit.